src/cli/templates/_worker.js/index.ts

```diff
@@ -0,0 +1,58 @@
+import { getCloudflareGlobalContextAls } from '../../../utils/utils/cloudflareGlobalContext';
+import { handleRequest } from './handleRequest';
+import { adjustRequestForVercel } from './utils';
+import type { AsyncLocalStorage } from 'node:async_hooks';
+
+declare const __NODE_ENV__: string;
+
+declare const __CONFIG__: ProcessedVercelConfig;
+
+declare const __BUILD_OUTPUT__: VercelBuildOutput;
+
+declare const __ENV_ALS_PROMISE__: Promise<null | AsyncLocalStorage<unknown>>;
+
+export default {
+	async fetch(request, env, ctx) {
+		const envAsyncLocalStorage = await __ENV_ALS_PROMISE__;
+		if (!envAsyncLocalStorage) {
+			return new Response(
+				`Error: Could not access built-in Node.js modules. Please make sure that your Cloudflare Pages project has the 'nodejs_compat' compatibility flag set.`,
+				{ status: 503 }
+			);
+		}
+
+		const cloudflareGlobalContextAls = getCloudflareGlobalContextAls();
+
+		if (!cloudflareGlobalContextAls) {
+			return new Response(
+				`Internal Server Error: cannot retrieve the cloudflare global context AsyncLocalStorage`,
```

Comment:
this should actually never occur, but I guess I think this makes the logic here a bit more robust

if preferred I don't mind using a non-null assertion on `cloudflareGlobalContextAls` or similar